### PR TITLE
Add docs for automatic service rollback

### LINKED ISSUE #2754: Document service rollback
### Problem description

Document service rollback feature https://github.com/docker/docker/pull/31108

### Project version(s) affected

This feature is in 17.04.x and will be stable with 17.06.

### PR DESCRIPTION
**Pull request by mstanleyjones**
### Proposed changes

Document automatic rollback of services
Fixes #2754 

@aaronlehmann PTAL

### CODE REVIEW

File: engine/swarm/services.md [hunk 1]
```diff
@@ -522,7 +522,50 @@ $ docker service update \
...
 
 my_web
+```
+
+In Docker 17.04 and higher, you can configure a service to roll back
+automatically if a service update fails to deploy. See
+[Automatically roll back if an update fails](#automatically-roll-back-if-an-update-fails).
+
+Related to the new automatic rollback feature, in Docker 17.04 and higher,
+manual rollback is handled at the service side, rather than the client, if the
```
Comment (aaronlehmann) [inline engine/swarm/services.md]: "server side"

File: engine/swarm/services.md [hunk 2]
```diff
@@ -522,7 +522,50 @@ $ docker service update \
...
+set a value, the default is used.
+
+| Flag                         | Default | Description                                   |
+|------------------------------|---------|-----------------------------------------------|
+|`--rollback-delay`            | `0s`    | Amount of time to wait after rolling back a task before rolling back the next one. A value of `0` means to roll back the second task immediately after the first rolled-back task deploys. |
+|`--rollback-failure-action`   | `pause` | When a task fails to roll back, whether to `pause` or `continue` trying to roll back other tasks. |
+|`--rollback-max-failure-ratio`| `0`     | The failure rate to tolerate during a rollback, specified as a floating-point number between 0 and 1. For instance, given 5 tasks, a failure ratio of `.2` would tolerate one task failing to roll back. A value of `0` means no failure are tolerated, while a value of `1` means any number of failure are tolerated. |
+|`--rollback-monitor`          | `0s`    | Duration after each task rollback to monitor for failure. If a task stops before this time period has elapsed, the rollback is considered to have failed. The default value disables monitoring. |
```
Comment (aaronlehmann) [inline engine/swarm/services.md]: The default is `5s`.

**Comment by mstanleyjones** [comment]
Addressed feedback.

**Comment by aaronlehmann** [comment]
LGTM

**Pull request merged**